core/pkg/sync/blob/blob_sync.go

```diff
@@ -131,6 +135,20 @@ func (hs *Sync) fetchObject(ctx context.Context, bucket *blob.Bucket) (string, e
 	if err != nil {
 		return "", fmt.Errorf("error downloading object %s/%s: %w", hs.Bucket, hs.Object, err)
 	}
+	if hs.fileType == "" {
+		uriSplit := strings.Split(hs.Object, ".")
+		hs.fileType = uriSplit[len(uriSplit)-1]
+	}
 
-	return buf.String(), nil
+	switch hs.fileType {
+	case "yaml", "yml":
+		return utils.YAMLToJSON(buf.Bytes())
+	case "json":
+		return buf.String(), nil
+	default:
+		// TODO make consistent with the file sync
+		return buf.String(), nil
+		// return "", fmt.Errorf("filepath extension for URI: '%s' is not supported", hs.sync())
```

Comment:
This retains the current behavior but is inconsistent with the file sync implementation. I'd prefer to return an error if the file extension is invalid or undefined but it would be a breaking change.

Do not merge until a decision has been made on the expected behavior.

Reply:
I agree that throwing here would be a good idea. But we could also do a slow transition, and for now, only log an error and a deprecation warning that the current file is not supported, and in future versions, this will cause an error. wdyt?

Reply:
I think we should just add the break. We are still sub zero and I don't think it will effect many people, and the resolution is very clear.